content/blog/rise-and-rise-of-graphql/index.md

```diff
@@ -0,0 +1,94 @@
+
+## Why to use GraphQL?
+
+GraphQL **offers much power and flexibility** 🦄 over Traditional APIs. One of the few things GraphQL was designed, was to solve Data Fetching/Loading problems.
+
+Data fetching is one of the most critical factors responsible for the performance of the application. Most of the time, the data provided by the API isn't required to run the application. This problem is termed as **"Over fetching"**, which can be a bigger issue in mobile devices. Since devices with slower network connection are bound to take more load time, it creates a **negative impact on end-users.**
+
+The current scenario where traditional REST APIs offer no clean way to solve this problem, GraphQL comes to the rescue. It allows **clients to define the structure of the data** required, and the same structure of the data is returned from the server. Therefore preventing fetching loads and loads of data, ultimately increasing performance.
+
+API endpoints generally fetch specific data, hence to load a certain data-rich application there has to be **multiple requests to the server.** But, applications should fetch relevant data in one round trip, to avoid **degrading performance.** The flexibility and richness of the GraphQL, allows us to define all the data to fetch, in a single request, avoiding multiple Round trips
+
+![Wrapping a REST API in GraphQL](./wrapper.png)
+_Wrapping REST API in graphQL - Source - Joey Ng'ethe | TwigaTech_
+
+## Fundamentals of GraphQL
+
+| Server -> | GraphQL Server -> | Graphql Client -> | Client |
+| --------- | ----------------- | ----------------- | ------ |
+
+
+---
+
+**Schema**
+GraphQL follows type system to define the schema of an API. The syntax for writing schemas is called Schema Definition Language (SDL).
+
+```js
+# A simple example to define schema
+type Person {
+  name: String!
+  age: Int!
+}
+```
+
+**Query**
+The data or information that a client needs from a server, can be fetched in a request using queries.
+
+```json
+# This query fetches all specifically names of all users
+{
+  users {
+    name
+  }
+}
+# This query fetches all names of all friends of user "Anup".
+{
+  user(name: "Anup") {
+    name
+    friend {
+	    name
+    }
+  }
+}
+```
+
+**Mutation**
+Mutations are the way to modify data to the server; these include creating/updating/deleting data. They have similar syntax as queries, with special keyword ahead of it.
+
+```json
+# This mutation adds a user with name "Anup" and age 20.
+mutation {
+  addUser( name : "Anup", age: 20) {
+    name
+    age
+  }
+}
+```
+
+## Cons
+
+Despite its powers, you might not need graphQL in your applications -
+
+- Since it adds unnecessary overhead and complex architecture for simple Applications
```

Comment:
please change Applications to applications. Rest everything looks good

Reply:
Done   👍

Reply:
Thanks